fc-community-contracts/src/FactChainSFT.sol

```diff
@@ -0,0 +1,80 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.20;
+
+import {ERC1155URIStorage} from "openzeppelin-contracts/contracts/token/ERC1155/extensions/ERC1155URIStorage.sol";
+import {ERC1155} from "openzeppelin-contracts/contracts/token/ERC1155/ERC1155.sol";
+import {Ownable} from "./utils/Ownable.sol";
+
+import {Arrays} from "openzeppelin-contracts/contracts/utils/Arrays.sol";
+
+interface IFactChainSFTEvents {
+    event FactchainNFTContractUpdated(address factchainNFTContract);
+    event FactchainBuildersRewarded(uint256 amount);
+    event CreatorRewarded(address creator, uint256 amount);
+}
+
+interface IFactChainSFT is IFactChainSFTEvents {
+    // Errors
+    error SupplyExhausted();
+    error ValueError();
+    error BadMintPrice();
+    error NegativeBalance();
+    error FailedToReward();
+    error ReservedToFactChain();
+}
+
+contract FactChainSFT is Ownable, ERC1155URIStorage, IFactChainSFT {
```

Comment:
Let's change all the `FactChain` to `Factchain`

Reply:
done